src/Storage/TSDatabaseView.m

```diff
@@ -265,11 +266,10 @@ + (void)asyncRegisterSecondaryDevicesDatabaseView
 + (NSDate *)localTimeReceiveDateForInteraction:(TSInteraction *)interaction {
     NSDate *interactionDate = interaction.date;
 
-    if ([interaction isKindOfClass:[TSIncomingMessage class]]) {
-        TSIncomingMessage *message = (TSIncomingMessage *)interaction;
-
-        if (message.receivedAt) {
-            interactionDate = message.receivedAt;
+    if ([interaction isKindOfClass:[TSMessage class]]) {
```

Comment:
How are we going to sort existing messages by bestReceivedAt if it's nil?

Two alternatives:
- The default implementation of `bestReceivedAtDate` could return `interaction.date` if `receivedAt` is nil.
OR
- Do the same thing, but in the NSCoder initializer

As is, it might work for now, but I suspect the next time we bump the version on this view we'll lose all sorting for the old messages.

Though you've done a good job of commenting the properties, I'd lean towards migrating the data, so that down the road we're less likely to be surprised by a nil field that "should never be nil".